src/renderer/viz/expressions/histogram/Histogram.js

```diff
@@ -111,4 +104,93 @@ export default class Histogram extends BaseExpression {
             };
         });
     }
+
+    _getDateValue (histogram, size) {
+        const array = [...histogram].map((value) => {
+            return { x: value[0].getTime(), y: value[1] };
+        });
+        const arrayLength = array.length;
+
+        let min = Number.POSITIVE_INFINITY;
+        let max = Number.NEGATIVE_INFINITY;
+
+        for (let i = 0; i < arrayLength; i++) {
+            const x = array[i].x;
+            min = Math.min(min, x);
+            max = Math.max(max, x);
+        }
+
+        const hist = Array(size).fill(0);
+        const range = max - min;
+        const sizeMinusOne = size - 1;
+
+        for (let i = 0; i < arrayLength; i++) {
+            const x = array[i].x;
+            const y = array[i].y;
+            const index = Math.min(Math.floor(size * (x - min) / range), sizeMinusOne);
+            hist[index] += y;
+        }
+
+        return hist.map((count, index) => {
+            const x0 = new Date(min + index / size * range);
+            const x1 = new Date(min + (index + 1) / size * range);
+
+            return {
+                x: [x0, x1],
+                y: count
+            };
+        });
+    }
+
+    _genBreakpoints () {
+        const histogram = this._histogram.value;
+
+        if (!histogram) {
+            return;
+        }
+
+        const accumHistogram = this._getAccumHistogramFrom(histogram);
+        const [min, max] = this._getMinMaxFrom(histogram);
+
+        this._updateBreakpointsWith({ accumHistogram, min, max });
+    }
+
+    _getAccumHistogramFrom (histogram) {
+        let prev = 0;
+        const accumHistogram = histogram.map(({ y }) => {
+            prev += y;
+            return prev;
+        });
+        return accumHistogram;
+    }
+
+    _getMinMaxFrom (histogram) {
+        const min = histogram[0].x[0];
+        const max = histogram[histogram.length - 1].x[1];
+
+        return [min, max];
+    }
+
+    _sortNumerically (a, b) {
+        const frequencyDifference = (b.y - a.y);
+
+        if (frequencyDifference === 0) {
+            const categoryA = a.x;
+            const categoryB = b.x;
+
+            if (!categoryA && !categoryB) { return 0; } // both null or undefined
+            if (!categoryA) { return 1; } // categoryB first
+            if (!categoryB) { return -1; } // categoryA first
+
+            if (typeof categoryA === 'string' && typeof categoryB === 'string') {
```

Comment:
How about adding a small explaining comment? It feels a bit strange to '_sortNumerically' en then dealing with 'string'

Reply:
This is how it was before, I can change it to make it more meaningful.